### PR TITLE
Support webpack

### LINKED ISSUE #96: fastify-plugin cannot be bundle
## 🐛 Bug Report

After that change https://github.com/fastify/fastify-plugin/pull/94 I can't bundle fastify using Webpack because `require.main.filename` is `undefined`.

## To Reproduce

- Download [fastify-webpack.zip](https://github.com/fastify/fastify-plugin/files/5021080/fastify-webpack.zip)
- `npm ci`
- `npm run build`
- `node .`
- And you'll see: `fastify not found, proceeding anyway`

## Expected behavior

Fastify plugin should handle the case where `require.main.filename` is `undefined`.

## Possible solution

I think it'll would be better to mix the previous solution and new one.
Something like that should do the trick:

```js
if (require.main.filename) {
  const pkgPath = join(dirname(require.resolve('fastify', { paths: [require.main.filename] })), 'package.json')
  fastifyVersion = semver.coerce(require(pkgPath).version)
} else {
  fastifyVersion = semver.coerce(require('fastify/package.json').version)
}
```

## Your Environment

- *node version*: 10
- *fastify version*: 3.1.1
- *os*: Windows

### PR DESCRIPTION
**Pull request by mcollina**
Fix #96

#### Checklist

- [x] run `npm run test` and `npm run benchmark`
- [x] tests and/or benchmarks are included
- [x] commit message and code follows [Code of conduct](https://github.com/fastify/fastify/blob/master/CODE_OF_CONDUCT.md)

### CODE REVIEW

**Comment by mcollina** [comment]
cc @Johann-S

File: test/bundlers.test.js [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+  const info = console.info
+  t.tearDown(() => {
+    require.main.filename = filename
+    console.info = info
+  })
+
+  require.main.filename = null
+
+  console.info = function (msg) {
+    t.fail('logged: ' + msg)
```
Comment (jsumners) [inline test/bundlers.test.js:17]: Shouldn't `t.end()` be within this function?

**Comment by Johann-S** [comment]
Thanks @mcollina 👍

File: test/bundlers.test.js (hunk 1 above)
Comment (mcollina) [reply test/bundlers.test.js:17]: This will automatically fail the test.

Comment (jsumners) [reply test/bundlers.test.js:17]: Oh. I mis-read the test.

🤔 shouldn't there at least be a `t.pass` somewhere in this test?

Comment (mcollina) [reply test/bundlers.test.js:17]: Maybe, there is not really a need.

**Pull request merged**